sandbox/host/src/sandbox/wasmi_backend.rs

```diff
@@ -18,393 +18,544 @@
-            ))
-        })?;
-
-        Ok(sandbox_wasmi::FuncInstance::alloc_host(
-            signature.clone(),
-            idx.0,
-        ))
+fn into_value(value: &wasmi::Val) -> Option<Value> {
+    match value {
+        wasmi::Val::I32(val) => Some(Value::I32(*val)),
+        wasmi::Val::I64(val) => Some(Value::I64(*val)),
+        wasmi::Val::F32(val) => Some(Value::F32((*val).into())),
+        wasmi::Val::F64(val) => Some(Value::F64((*val).into())),
+        _ => None,
     }
+}
+
+/// Wasmi specific context
+pub struct Backend {
+    store: Rc<StoreRefCell>,
+}
 
-    fn resolve_memory(
-        &self,
-        module_name: &str,
-        field_name: &str,
-        _memory_type: &sandbox_wasmi::MemoryDescriptor,
-    ) -> std::result::Result<sandbox_wasmi::MemoryRef, sandbox_wasmi::Error> {
-        let mem = self
-            .memory_by_name(module_name, field_name)
-            .ok_or_else(|| {
-                sandbox_wasmi::Error::Instantiation(format!(
-                    "Export {}:{} not found",
-                    module_name, field_name
-                ))
-            })?;
-
-        let wrapper = mem.as_wasmi().ok_or_else(|| {
-            sandbox_wasmi::Error::Instantiation(format!(
-                "Unsupported non-wasmi export {}:{}",
-                module_name, field_name
-            ))
-        })?;
-
-        // Here we use inner memory reference only to resolve the imports
-        // without accessing the memory contents. All subsequent memory accesses
-        // should happen through the wrapper, that enforces the memory access protocol.
-        let mem = wrapper.0;
-
-        Ok(mem)
+impl Default for Backend {
+    fn default() -> Self {
+        Self::new()
     }
+}
 
-    fn resolve_global(
-        &self,
-        module_name: &str,
-        field_name: &str,
-        _global_type: &sandbox_wasmi::GlobalDescriptor,
-    ) -> std::result::Result<sandbox_wasmi::GlobalRef, sandbox_wasmi::Error> {
-        Err(sandbox_wasmi::Error::Instantiation(format!(
-            "Export {}:{} not found",
-            module_name, field_name
-        )))
+impl Backend {
+    pub fn new() -> Self {
+        let register_len = size_of::<UntypedVal>();
+
+        const DEFAULT_MIN_VALUE_STACK_HEIGHT: usize = 1024;
+        const DEFAULT_MAX_VALUE_STACK_HEIGHT: usize = 1024 * DEFAULT_MIN_VALUE_STACK_HEIGHT;
+        const DEFAULT_MAX_RECURSION_DEPTH: usize = 16384;
+
+        let mut config = Config::default();
+        config.set_stack_limits(
+            StackLimits::new(
+                DEFAULT_MIN_VALUE_STACK_HEIGHT / register_len,
+                DEFAULT_MAX_VALUE_STACK_HEIGHT / register_len,
+                DEFAULT_MAX_RECURSION_DEPTH,
+            )
+            .expect("infallible"),
+        );
+
+        let engine = Engine::new(&config);
+        let store = Store::new(&engine, None);
+        Backend {
+            store: Rc::new(StoreRefCell::new(store)),
+        }
```

Comment:
Original code have never set the config. It only used in embedded executor specifically for fuzzer

Reply:
Fixed